components/raftstore/src/store/peer.rs

```diff
@@ -3595,9 +3595,32 @@ where
         ctx.raft_metrics.propose.transfer_leader += 1;
 
         let transfer_leader = get_transfer_leader_cmd(&req).unwrap();
-        let peer = transfer_leader.get_peer();
+        let peers = transfer_leader.get_peers();
+
+        let transferred = match peers.is_empty() {
+            true => self.pre_transfer_leader(transfer_leader.get_peer()),
+            false => {
+                let prs = self.raft_group.raft.prs();
+                let mut peer = metapb::Peer::default();
+                let mut max_matched = 0;
+
+                for candidate in peers {
```

Comment:
You mean `peers.iter().max_by_key(|c| prs.get(c.id).map_or(0, |pr| pr.matched))`?